consensus/polybft/extra_test.go

```diff
@@ -0,0 +1,484 @@
+
+	copiedDelta := originalDelta.Copy()
+	require.NotNil(t, copiedDelta)
+	require.NotSame(t, originalDelta, copiedDelta)
+	require.NotEqual(t, originalDelta, copiedDelta)
+	require.Empty(t, copiedDelta.Added)
+	require.Equal(t, copiedDelta.Removed.Len(), originalDelta.Removed.Len())
+
+	newValidators := newTestValidators(addedValidatorsCount).getPublicIdentities()
+	copiedDelta.Added = append(copiedDelta.Added, newValidators...)
+	require.Empty(t, originalDelta.Added)
+	require.Len(t, copiedDelta.Added, addedValidatorsCount)
+	require.Equal(t, copiedDelta.Removed.Len(), originalDelta.Removed.Len())
+}
+
+func TestValidatorSetDelta_UnmarshalRLPWith_NegativeCases(t *testing.T) {
+	t.Run("Incorrect RLP value type provided", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		delta := &ValidatorSetDelta{}
+		require.ErrorContains(t, delta.UnmarshalRLPWith(ar.NewNull()), "value is not of type array")
+	})
+
+	t.Run("Empty RLP array provided", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		delta := &ValidatorSetDelta{}
+		require.NoError(t, delta.UnmarshalRLPWith(ar.NewArray()))
+	})
+
+	t.Run("Incorrect RLP array size", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		deltaMarshalled := ar.NewArray()
+		deltaMarshalled.Set(ar.NewBytes([]byte{0x59}))
+		deltaMarshalled.Set(ar.NewBytes([]byte{0x33}))
+		deltaMarshalled.Set(ar.NewBytes([]byte{0x26}))
+		delta := &ValidatorSetDelta{}
+		require.ErrorContains(t, delta.UnmarshalRLPWith(deltaMarshalled), "incorrect elements count to decode validator set delta, expected 2 but found 3")
+	})
+
+	t.Run("Incorrect RLP value type for Added field", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		deltaMarshalled := ar.NewArray()
+		deltaMarshalled.Set(ar.NewBytes([]byte{0x59}))
+		deltaMarshalled.Set(ar.NewBytes([]byte{0x33}))
+		delta := &ValidatorSetDelta{}
+		require.ErrorContains(t, delta.UnmarshalRLPWith(deltaMarshalled), "array expected for added validators")
+	})
+
+	t.Run("Incorrect RLP value type for ValidatorAccount in Added field", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		deltaMarshalled := ar.NewArray()
+		addedArray := ar.NewArray()
+		addedArray.Set(ar.NewNull())
+		deltaMarshalled.Set(addedArray)
+		deltaMarshalled.Set(ar.NewNull())
+		delta := &ValidatorSetDelta{}
+		require.ErrorContains(t, delta.UnmarshalRLPWith(deltaMarshalled), "value is not of type array")
+	})
+
+	t.Run("Incorrect RLP value type for Removed field", func(t *testing.T) {
+		ar := &fastrlp.Arena{}
+		deltaMarshalled := ar.NewArray()
+		addedValidators := newTestValidators(3).getPublicIdentities()
+		addedArray := ar.NewArray()
+		for _, validator := range addedValidators {
+			addedArray.Set(validator.MarshalRLPWith(ar))
+		}
+		deltaMarshalled.Set(addedArray)
+		deltaMarshalled.Set(ar.NewNull())
+		delta := &ValidatorSetDelta{}
+		require.ErrorContains(t, delta.UnmarshalRLPWith(deltaMarshalled), "value is not of type bytes")
+	})
+}
+
+func Test_GetIbftExtraClean_Fail(t *testing.T) {
```

Comment:
Test can be parallelized with `t.Parallel()`

Reply:
Fixed through #825